internal/context/context.go

```diff
@@ -0,0 +1,54 @@
+package context
+
+import (
+	stdcontext "context"
+)
+
+type (
+	// Context augments the context.Context interface with addition methods
+	// to allow to retrieve the extra attributes our wrapper context holds
+	Context interface {
+		stdcontext.Context
+		Method() string
+		Arguments() map[string]interface{}
+		CorrelationID() string
+	}
+	// context wraps context.Context allowing adding tracing information instead
+	// of using the Values.
+	context struct {
```

Comment:
Does it make sense to make it compatible with the internal context?

Reply:
@superdecimal it is compatible :)

Reply:
You can use this as a normal context.Context